api/v1alpha1/stardoginstance_types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// StardogInstanceSpec defines the desired state of StardogInstance
+type StardogInstanceSpec struct {
+	// ServerUrl describes the url of the Stardog Instace
+	ServerUrl string `json:"serverUrl,omitempty"`
+	// AdminCredentials describes credentials that gives access to the Stardog Instace
+	AdminCredentials StardogAdminCredentialsSpec `json:"adminCredentials,omitempty"`
+}
+
+// StardogAdminCredentialsSpec specifies the password of the Stardog admin user
+type StardogAdminCredentialsSpec struct {
+	// SecretRef describes a reference to a secret where the credential is contained
+	SecretRef corev1.SecretKeySelector `json:"secretRef,omitempty"`
+}
```

Comment:
We can reuse the `StardogUserCredentialsSpec`, no need for another struct that is basically the same
```suggestion
	// AdminCredentials references the credentials that gives administrative access to the Stardog instance.
	// This is used by the Operator to make changes in the roles, permissions and users.
	AdminCredentials StardogUserCredentialsSpec `json:"adminCredentials,omitempty"`
}

// StardogAdminCredentialsSpec specifies the password of the Stardog admin user
type StardogAdminCredentialsSpec struct {
	// SecretRef describes a reference to a secret where the credential is contained
	SecretRef corev1.SecretKeySelector `json:"secretRef,omitempty"`
}
```
`StardogAdminCredentialsSpec` can be deleted